chain-impl-mockchain/src/leadership/bft.rs

```diff
@@ -105,6 +106,18 @@ impl From<PublicKey<SIGNING_ALGORITHM>> for LeaderId {
     }
 }
 
+impl Bech32 for LeaderId {
```

Comment:
I am not sure this is wise to start implementing this trait for LeaderId. There is no reason to believe this type and a public key are both the same object. The HRP here is not necessarily true.